src/pymatgen/io/vasp/outputs.py

```diff
@@ -2106,19 +2106,14 @@ def __init__(self, filename: PathLike) -> None:
         self.drift = self.data.get("drift", [])
 
         # Check if calculation is spin polarized
-        self.spin = False
-        self.read_pattern({"spin": "ISPIN  =      2"})
-        if self.data.get("spin", []):
-            self.spin = True
+        self.read_pattern({"spin": r"ISPIN\s*=\s*2"}, terminate_on_match=True)
+        self.spin = bool(self.data.get("spin", []))
 
         # Check if calculation is non-collinear
-        self.noncollinear = False
-        self.read_pattern({"noncollinear": "LNONCOLLINEAR =      T"})
-        if self.data.get("noncollinear", []):
-            self.noncollinear = False
+        self.read_pattern({"noncollinear": r"LNONCOLLINEAR\s*=\s*T"}, terminate_on_match=True)
```

Comment:
these lines suggest that the `matches` dict can contain less keys if `terminate_on_match=True` and so you might get a different `self.data` after the update. but given all tests pass, this may not happen in practice
https://github.com/materialsproject/pymatgen/blob/edcd4654cc5048d7f0f00417d82755d98b03b739/src/pymatgen/io/pwscf.py#L623-L630

Reply:
any concerns @mkhorton?